src/encdatavault_fmt_plug.c

```diff
@@ -0,0 +1,434 @@
+	buffer_128 tmp_out = { 0 };
+	int i, j;
+
+	AES_set_encrypt_key(key, ENC_KEY_SIZE * 8, &aes_key);
+	len = len >> 4;
+	counter = counter << 56;
+
+	for (i = 0; i < len; i++) {
+		tmp_iv.u64[0] = ivs[0].u64[0];
+		tmp_iv.u64[1] = counter;
+		AES_encrypt(tmp_iv.u8, tmp_iv.u8, &aes_key);
+		memcpy(tmp_out.u8, in, AES_BLOCK_SIZE);
+		enc_xor_block(tmp_out.u64, tmp_iv.u64);
+
+		for (j = 1; j < nb_keys; j++) {
+			tmp_iv.u64[0] = ivs[j].u64[0];
+			tmp_iv.u64[1] = counter;
+			AES_encrypt(tmp_iv.u8, tmp_iv.u8, &aes_key);
+			enc_xor_block(tmp_out.u64, tmp_iv.u64);
+		}
+		memcpy(out, tmp_out.u8, AES_BLOCK_SIZE);
+
+		// Increment counter, only 255 block encryption is supported.
+		counter += 0x0100000000000000;
+		out += AES_BLOCK_SIZE;
+		in += AES_BLOCK_SIZE;
+	}
+}
+
+static void init(struct fmt_main *self)
+{
+	omp_autotune(self, OMP_SCALE);
+	saved_key =
+	    mem_calloc(sizeof(*saved_key), self->params.max_keys_per_crypt);
+	cracked = mem_calloc(sizeof(*cracked), self->params.max_keys_per_crypt);
+	cracked_size = sizeof(*cracked) * self->params.max_keys_per_crypt;
+	any_cracked = 0;
+}
+
+static void done(void)
+{
+	MEM_FREE(saved_key);
+	MEM_FREE(cracked);
+}
+
+static int valid(char *ciphertext, struct fmt_main *self)
+{
+	char *p = ciphertext, *ctcopy, *keeptr;
+	int extra;
+
+	if (strncmp(ciphertext, FORMAT_TAG, FORMAT_TAG_LENGTH) != 0)
+		return 0;
+
+	ctcopy = strdup(ciphertext);
+	keeptr = ctcopy;
+	ctcopy += FORMAT_TAG_LENGTH;
+	if ((p = strtokm(ctcopy, "$")) == NULL) // version
+		goto err;
+	if (!isdec(p))
+		goto err;
+	if (strcmp(p, "1") && strcmp(p, "3")) {
+		static int warned;
+
+		if (!warned++)
+			fprintf(stderr,
+			        "%s: Warning: version %d not supported, not loading such hashes!\n",
+			        self->params.label, atoi(p));
+		goto err;
+	}
+	if ((p = strtokm(NULL, "$")) == NULL)   // algorithm id
+		goto err;
+	if (!isdec(p))
+		goto err;
+	if (atoi(p) > 4 || atoi(p) < 1) {
+		static int warned;
+
+		if (!warned++)
+			fprintf(stderr, "%s: Warning: algorithm id %d not supported!\n",
+			        self->params.label, atoi(p));
+		goto err;
+	}
+	if ((p = strtokm(NULL, "$")) == NULL)   // Nonce
+		goto err;
+	if (hexlenl(p, &extra) != ENC_NONCE_SIZE * 2 || extra)
+		goto err;
+	if ((p = strtokm(NULL, "$")) == NULL)   // Encrypted header
+		goto err;
+	if (hexlenl(p, &extra) != ENC_SIG_SIZE * 2 || extra)
+		goto err;
+	MEM_FREE(keeptr);
+	return 1;
+
+err:
+	MEM_FREE(keeptr);
+	return 0;
+}
+
+static void *get_salt(char *ciphertext)
+{
+	static struct custom_salt cs;
+	int i;
+	char *p = ciphertext, *ctcopy, *keeptr;
+
+	memset(&cs, 0, sizeof(cs));
+	ctcopy = strdup(ciphertext);
+	keeptr = ctcopy;
+	ctcopy += FORMAT_TAG_LENGTH;
+	p = strtokm(ctcopy, "$");
+	cs.version = atoi(p);
+	p = strtokm(NULL, "$");
+	cs.algo_id = atoi(p);
+	p = strtokm(NULL, "$");
+	for (i = 0; i < ENC_NONCE_SIZE; i++)
+		cs.iv[i] =
+		    (atoi16[ARCH_INDEX(p[2 * i])] << 4) | atoi16[ARCH_INDEX(p[2 * i +
+		            1])];
```

Comment:
BTW we've relaxed our code style from max. width of 80, to 120. Perhaps we should update `README.coding-style`.

Reply:
I have used the commands in `CONTRIBUTING.md`:
```bash
$ indent -kr -i4 -ts4 -nlp -nbbo -ncs -l79 -lc79 -bad -il0
$ astyle --style=kr -t4 -U -H -p -xC79 -c -k3 -z2
```

Reply:
@magnumripper has opened #4960 to fix that. You can change the 79 to 120 or 131.